lib/internal/util/comparisons.js

```diff
@@ -173,6 +173,11 @@ function innerDeepEqual(val1, val2, strict, memos) {
     return false;
   }
 
+  // Handle WeakMap and WeakSet
+  if (val1Tag === '[object WeakMap]' || val1Tag === '[object WeakSet]') {
+    return val1 === val2;
```

Comment:
The equality check happened above, so it's possible to directly return false.

Let's move this check down to the very last if statement about types. That way the performance of other types won't be impacted.